src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -57,20 +56,35 @@ public BtConnectorHelper() {
         this.myPeerName = peerName;
         this.lastAvailableList.clear();
         Stop();
-        mBTConnector = new BTConnector(context,this,this,conSettings,instanceEncryptionPWD);
+        mBTConnector = new BTConnector(context,this,this,conSettings);
         return mBTConnector.Start(this.myPeerIdentifier,this.myPeerName);
     }
 
+    public boolean isRunning(){
+        if(mBTConnector != null){
+            return true;
+        }else{
+            return false;
+        }
+    }
     public void Stop(){
         if(mBTConnector != null){
```

Comment:
Can we guarantee that this call will only be called from a single thread? Otherwise the call to stop on line 66 will result in a null exception. We should probably change this to store mBTConnector in a local variable before calling stop. This assumes, of course, that BTConnector itself is thread safe.